### DISCUSSION TITLE
Usage of <NuxtLink /> in Nuxt Server Components

### DISCUSSION THREAD

**Discussion by gxres042** [nuxt/nuxt#28318] [Questions] [answered]
I would like to try out Nuxt Server Components for my blog, however, <NuxtLink /> is a client-side navigation component, and using it within a server-side component would lose the refresh-free navigation feature.

Is there any way to use <NuxtLink /> within a server-side component and retain the refresh-free navigation feature?

**Comment by bernhardberger** [comment]
@gxres042 
https://roe.dev/blog/nuxt-server-components#:~:text=For%20now%2C%20%3CNuxtLink%3E%20components%20are%20not%20interactive%2C%20meaning%20you%20might%20need%20to%20add%20some%20code%20like%20this%20in%20the%20parent%20page%2C%20as%20a%20%27pretend%27%20version%20of%20client%2Dside%20routing%3A

You'll have to intercept the link click and trigger client side navigation from outside the component.

**Reply by gxres042** [reply]
> @gxres042 
> https://roe.dev/blog/nuxt-server-components#:~:text=For%20now%2C%20%3CNuxtLink%3E%20components%20are%20not%20interactive%2C%20meaning%20you%20might%20need%20to%20add%20some%20code%20like%20this%20in%20the%20parent%20page%2C%20as%20a%20%27pretend%27%20version%20of%20client%2Dside%20routing%3A
> 
> You'll have to intercept the link click and trigger client side navigation from outside the component. 

I've read this post, but I want a way to use <NuxtLink /> in server components directly like roe's blog.

**Reply by bernhardberger** [reply, marked as answer]
There is a new experimental feature "selectiveClient" since Nuxt 3.9:
https://nuxt.com/docs/guide/directory-structure/components#client-components-within-server-components

**Reply by gxres042** [reply]
> There is a new experimental feature "selectiveClient" since Nuxt 3.9:
> https://nuxt.com/docs/guide/directory-structure/components#client-components-within-server-components

That's a great idea

**Reply by gxres042** [reply]
> There is a new experimental feature "selectiveClient" since Nuxt 3.9: https://nuxt.com/docs/guide/directory-structure/components#client-components-within-server-components

Now I've try it like:

``` vue
// AComponent.server.vue
<NuxtLink nuxt-client to="/">Link</NuxtLink>
```

But it still lose the refresh-free navigation feature. What should I do now?

My Nuxt config:

``` typescript
export default defineNuxtConfig({
    	experimental: {
		componentIslands: {
			selectiveClient: true
		}
	},
})
```

**Reply by bernhardberger** [reply]
You need to add a `nuxt-client` directive to your NuxtLink, like so: `<NuxtLink nuxt-client href=..>`

**Reply by gxres042** [reply]
> You need to add a `nuxt-client` directive to your NuxtLink, like so: `<NuxtLink nuxt-client href=..>`

I'm surely I've added

![image](https://github.com/user-attachments/assets/f9052ab0-aa3c-4d9a-9e58-ae94cb0f384e)

andd when I set `experimental.componentIsland.selectiveClient` to `true`, rollup warned:

 WARN                                                                                                                                        20:45:58  
(!) Some chunks are larger than 500 kB after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.

**Reply by bernhardberger** [reply]
No idea tbh. My build fails completely when setting `selectiveClient` to `deep`.

I'm just relaying information here I found in the docs and changelogs.

Probably best to open an issue
